### DISCUSSION TITLE
Splitting FPrime project into separate application

### DISCUSSION THREAD

**Discussion by IonRod** [nasa/fprime#852] [Q&A] [answered]
Hi All!

I'm working on the robotic project on Linux. To make the system fault tolerant I'm thinking of splitting the project into several applications, for example separate app that controls robot's movement, separate app that controls video camera and so on.

All of the applications are initialised by systemd and work independently.

In case an app crashes it would be restarted by systemd.

What do you think of such a pattern?

**Comment by LeStarch** [comment, marked as answer]
I don't see anything wrong with this pattern per-se.  You could consider the hub pattern to allow your applications to communicate via ports between the processes.  This can facilitate inter-process communication in an F´ like way.

**Reply by LeStarch** [reply]
https://nasa.github.io/fprime/UsersGuide/best/hub-pattern.html

**Reply by LeStarch** [reply]
https://github.com/nasa/fprime/blob/devel/Svc/GenericHub/docs/sdd.md
